UI/ChatWnd.cpp

```diff
@@ -611,42 +613,104 @@ void MessageWnd::OpenForInput() {
 
 namespace {
     void SendChatMessage(const std::string& text, std::set<int> recipients) {
+        const ClientApp* app = ClientApp::GetApp();
+        if (!app) {
+            ErrorLogger() << "ChatWnd.cpp SendChatMessage couldn't get client app!";
+            return;
+        }
         ClientNetworking& net = HumanClientApp::GetApp()->Networking();
+        auto players = app->Players();
+        std::string recipient_names = "";
 
         if (recipients.empty()) {
             net.SendMessage(PlayerChatMessage(text));
         } else {
-            recipients.insert(HumanClientApp::GetApp()->PlayerID());   // ensure recipient sees own sent message
-            for (int recipient_id : recipients)
-                net.SendMessage(PlayerChatMessage(text, recipient_id));
+            for (int recipient_id : recipients) {
+                net.SendMessage(PlayerChatMessage(UserString("MESSAGES_WHISPER") + " " + text, recipient_id));
```

Comment:
(Whisper) probably shouldn't be baked into the actual message text. Rather, use a string format substutution on the recipient client when displaying

Reply:
Could you give an example where or how that is done?

Reply:
eg. https://github.com/freeorion/freeorion/blob/master/UI/CombatReport/GraphicalSummary.cpp#L358 or https://github.com/freeorion/freeorion/blob/master/UI/ChatWnd.cpp#L80 or search for `boost::io::str` for lots of examples.